source_code/src/NODEMGMT/node_mgmt.c

```diff
@@ -1330,3 +1330,32 @@ RET_TYPE updateChildNodeDescription(cNode* c, uint16_t cAddr, uint8_t* descripti
     return RETURN_OK;
 }
 
+/**
+ * Delete all data nodes child starting from firstChildNodeAddress
+ * @param dataNodeAddress     The address of the first child in the chain
+ * @return success status
+ * @note it is safe to call this function even with a NULL address (nothing is done)
+ */
+void deleteDataNodeChain(uint16_t dataNodeAddress)
```

Comment:
can you modify the function to take a pointer to a temp_dnode instead? (less stack usage)
temp_dnode_ptr will do fine